merlin/dag/executors.py

```diff
@@ -48,110 +48,114 @@ def transform(
         Transforms a single dataframe (possibly a partition of a Dask Dataframe)
         by applying the operators from a collection of Nodes
         """
-        output_df = None
+        output_data = None
 
         for node in nodes:
-            input_df = self._build_input_data(node, df, capture_dtypes=capture_dtypes)
+            input_data = self._build_input_data(node, transformable, capture_dtypes=capture_dtypes)
 
             if node.op:
-                transformed_df = self._transform_data(node, input_df, capture_dtypes=capture_dtypes)
+                transformed_data = self._transform_data(
+                    node, input_data, capture_dtypes=capture_dtypes
+                )
             else:
-                transformed_df = input_df
+                transformed_data = input_data
 
-            output_df = self._combine_node_outputs(node, transformed_df, output_df)
+            output_data = self._combine_node_outputs(node, transformed_data, output_data)
 
         if additional_columns:
-            output_df = concat_columns([output_df, df[_get_unique(additional_columns)]])
+            output_data = concat_columns(
```

Comment:
It seems like if this line were ever reached it would throw an exception in the case of the non-dataframe input data

Reply:
You’re right, it does! Have a fix for `concat_columns` that still needs to get pushed.

Reply:
Added a commit to fix this